include/private/rdydmimpl.h

```diff
@@ -0,0 +1,12 @@
+#ifndef RDYDMIMPL_H
+#define RDYDMIMPL_H
+
+#include <petsc.h>
+#include <rdycore.h>
+
+PETSC_INTERN PetscErrorCode CloneAndCreateCellCenteredDM(DM dm, PetscBool is_dm_refined, PetscInt n_aux_field, PetscInt n_aux_field_dof[n_aux_field],
```

Comment:
These declarations could also go in `include/private/rdycoreimpl.h`, for what it's worth, since they are internal operations that support the dycore. But it's not a big deal, and we can always reorganize things later when we need to.